tests/ulm/erc20/erc20_negative_test.sh

```diff
@@ -0,0 +1,473 @@
+  local key=$1
+  local contract=$2
+  local from_account=$3
+  local allowed_account=$4
+  call http://localhost:8545 erc20 $contract /dev/stdin 0 allowance $from_account $allowed_account <<< $key
+}
+
+function erc20_mint {
+  local key=$1
+  local contract=$2
+  local account=$3
+  local amount=$4
+  call http://localhost:8545 erc20 $contract /dev/stdin 0 mint $account $amount <<< $key
+}
+
+function erc20_transfer {
+  local key=$1
+  local contract=$2
+  local account=$3
+  local amount=$4
+  call http://localhost:8545 erc20 $contract /dev/stdin 0 transfer $account $amount <<< $key
+}
+
+function erc20_approve {
+  local key=$1
+  local contract=$2
+  local account=$3
+  local amount=$4
+  call http://localhost:8545 erc20 $contract /dev/stdin 0 approve $account $amount <<< $key
+}
+
+function erc20_transfer_from {
+  local key=$1
+  local contract=$2
+  local from_account=$3
+  local to_account=$4
+  local amount=$5
+  call http://localhost:8545 erc20 $contract /dev/stdin 0 transferFrom $from_account $to_account $amount <<< $key
+}
+
+function test_transfer_too_much {
+  echo -n "Transfer test "
+
+  # generate some accounts
+  account1=($(mkacct))
+  a1=${account1[0]}
+  k1=${account1[1]}
+  echo -n "."
+
+  account2=($(mkacct))
+  a2=${account2[0]}
+  k2=${account2[1]}
+  echo -n "."
+
+  account3=($(mkacct))
+  a3=${account3[0]}
+  k3=${account3[1]}
+  echo -n "."
+
+  # fund accounts
+  fund /dev/stdin <<< $k1
+  echo -n "."
+  fund /dev/stdin <<< $k2
+  echo -n "."
+  fund /dev/stdin <<< $k3
+  echo -n "."
+
+  # deploy contract
+  contract=$(erc20_deploy $k1)
+  echo -n "."
+
+  balance2=$(erc20_balanceOf $k1 $contract $a2)
+  echo -n "."
+  assert_eq "0" "$balance2" "Balance of Account 2"
+
+  balance3=$(erc20_balanceOf $k1 $contract $a3)
+  echo -n "."
+  assert_eq "0" "$balance3" "Balance of Account 3"
+
+  # check total supply
+  supply=$(erc20_totalSupply $k1 $contract)
+  echo -n "."
+  assert_eq "0" "$supply" "Total Supply"
+
+  erc20_mint $k1 $contract $a2 $(to_hex 1000)
+  echo -n "."
+
+  balance2=$(erc20_balanceOf $k1 $contract $a2)
+  echo -n "."
+  assert_eq "1000" "$balance2" "Balance of Account 2"
+
+  balance3=$(erc20_balanceOf $k1 $contract $a3)
+  echo -n "."
+  assert_eq "0" "$balance3" "Balance of Account 3"
+
+  # check total supply
+  supply=$(erc20_totalSupply $k1 $contract)
+  echo -n "."
+  assert_eq "1000" "$supply" "Total Supply"
+
+  erc20_transfer $k2 $contract $a3 $(to_hex 600)
+  echo -n "."
+
+  balance2=$(erc20_balanceOf $k1 $contract $a2)
+  echo -n "."
+  assert_eq "400" "$balance2" "Balance of Account 2"
+
+  balance3=$(erc20_balanceOf $k1 $contract $a3)
+  echo -n "."
+  assert_eq "600" "$balance3" "Balance of Account 3"
+
+  # check total supply
+  supply=$(erc20_totalSupply $k1 $contract)
+  echo -n "."
+  assert_eq "1000" "$supply" "Total Supply"
+
+  echo -e -n "\nTransfer too much test (should fail): "
+  erc20_transfer $k2 $contract $a3 $(to_hex 600) || true  # this should fail
```

Comment:
It might be nice to fail here if the call passes (my understanding is the call will always be considering passing with `|| true` appended) --- though it should not be possible for the call to pass here and for the remaining calls to still pass.